tofu/main/harvester/variables.tf

```diff
@@ -0,0 +1,159 @@
+  default     = "~/.ssh/id_ed25519"
+}
+
+variable "ssh_user" {
+  description = "User name to use for the SSH connection to all nodes in all clusters"
+  default     = "opensuse"
+}
+
+variable "ssh_bastion_host" {
+  description = "Public name of the SSH bastion host. Leave null for publicly accessible Harvester instances"
+  type        = string
+  default     = null
+}
+
+variable "ssh_bastion_user" {
+  description = "User name to connect to the SSH bastion host"
+  default     = null
+}
+
+variable "ssh_bastion_key_path" {
+  description = "Path of private ssh key used to access the bastion host to access Harvester"
+  type        = string
+  default     = null
+}
+
+variable "upstream_cluster" {
+  description = "Upstream cluster configuration. See tofu/modules/generic/test_environment/variables.tf for details"
+  type        = any
+}
+
+variable "upstream_cluster_distro_module" {
+  description = "Name of the module to use for the upstream cluster"
+  default     = "generic/k3s"
+}
+
+variable "downstream_cluster_templates" {
+  description = "List of downstream cluster configurations. See tofu/modules/generic/test_environment/variables.tf for details"
+  type        = list(any)
+}
+
+variable "downstream_cluster_distro_module" {
+  description = "Name of the module to use for the downstream clusters"
+  default     = "generic/k3s"
+}
+
+variable "tester_cluster" {
+  description = "Tester cluster configuration. See tofu/modules/generic/test_environment/variables.tf for details"
+  type        = any
+  default     = null
+}
+
+variable "tester_cluster_distro_module" {
+  description = "Name of the module to use for the downstream clusters"
+  default     = "generic/k3s"
+}
+
+# "Multi-tenancy" variables
+variable "project_name" {
+  description = "Name of this project, used as prefix for resources it creates"
+  default     = "st"
+}
+
+variable "first_kubernetes_api_port" {
+  description = "Port number where the Kubernetes API of the first cluster is published locally. Other clusters' ports are published in successive ports"
+  type        = number
+  default     = 7445
+}
+
+variable "first_app_http_port" {
+  description = "Port number where the first server's port 80 is published locally. Other clusters' ports are published in successive ports"
+  type        = number
+  default     = 9080
+}
+
+variable "first_app_https_port" {
+  description = "Port number where the first server's port 443 is published locally. Other clusters' ports are published in successive ports"
+  type        = number
+  default     = 9443
+}
+
+# Harvester-specific variables
+variable "namespace" {
+  description = "The namespace where the VMs should be created"
+  default     = "default"
+}
+
+variable "kubeconfig" {
+  description = "Path to the Harvester kubeconfig file. Uses KUBECONFIG by default. See https://docs.harvesterhci.io/v1.3/faq/#how-can-i-access-the-kubeconfig-file-of-the-harvester-cluster"
+  type        = string
+  default     = null
+}
```

Comment:
Just want to confirm -- this is not a "nullable" field, right? We want it to error by default if no var is provided here?

Reply:
So technically it -is- nullable because the provider should use the OS' `KUBECONFIG` env var if this is not explicitly set. But due to the way we are setting it directly, no it shouldn't be. Will update